### ISSUE TITLE
modifying a copy of a project changes the original

### ISSUE THREAD

**Issue by Raizo62** [GNS3/gns3-gui#3650]
Hi

I'm not sure if this is a bug.
I think I see where it's coming from, but it has a surprising effect.

- With GNS3 2.2, i create a new projet ("untitle") with devices.
- I close GNS3.

- With Linux, i do a copy of this project with the name "untitle_copy"
`cp -R untitle untitle_copy`

- With GNS3, i want to modify the project "untitle_copy".
- As in GNS3, in "Projects library", i don't see my copy, i force with the button "Open a project from disk" and i open it. (The title in GNS3 is "untitle")
- I move the devices on this project.
- I close GNS3.

- With GNS3, I open the project "untitle".
- Devices are also moved

**Comment by grossmj** [comment]
> With Linux, i do a copy of this project with the name "untitle_copy"
cp -R untitle untitle_copy

Hum, there is a project ID inside the .gns3 file, this means the project ID will be the same if you copy projects like that.

**Comment by Raizo62** [comment]
I think also. But it gives a strange feeling : we don't expect that

**Comment by grossmj** [comment]
Maybe we should check for duplicated project ID?

**Comment by Raizo62** [comment]
> Maybe we should check for duplicated project ID?

Okay, but then what do you do with your multiple project IDs ? How do you choose ?

**Comment by grossmj** [comment]
> Okay, but then what do you do with your multiple project IDs ? How do you choose ?

We could automatically generate a new project ID or warn users there are projects with duplicated IDs. Not sure it's worth doing anything...

**Comment by Raizo62** [comment]
I don't understand something: When we choose a project (example A), GNS3 goes through all the projects to find the ID of A?

**Comment by grossmj** [comment]
> I don't understand something: When we choose a project (example A), GNS3 goes through all the projects to find the ID of A?

No, the projects info is loaded by the controller. Then when you select your project, GNS3 will use its ID to find in the list.

**Comment by muriloglix** [comment]
> > Okay, but then what do you do with your multiple project IDs ? How do you choose ?
> 
> We could automatically generate a new project ID or warn users there are projects with duplicated IDs. Not sure it's worth doing anything...

I see three possibilities:
1. When importing the project, check for an existing ID and display a new window asking if the user wants to generate a new ID or cancel the operation.
2. When importing a project, automatically generate a new ID and continue the import process.
3. When importing a project, check if the ID already exists on the server. If found, generate a new ID and continue the import process.

I believe that the simplest would be option number 2, since, when importing, we can consider that it will be a "new project" on the server. This would simplify the code, since it would only generate a new ID, there would be no need to ask the client anything and there would be no need to check for duplicate IDs. Everything would work transparently for the user.
